_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-mailchimp.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * Mailchimp: Get Mailchimp Status.
+ * API to determine if current site has linked Mailchimp account and mailing list selected.
+ * This API is meant to be used in Jetpack and on WPCOM.
+ *
+ * @since 7.0
+ */
+class WPCOM_REST_API_V2_Endpoint_Mailchimp extends WP_REST_Controller {
+	public function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'mailchimp';
+		$this->wpcom_is_wpcom_only_endpoint = true;
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Called automatically on `rest_api_init()`.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base,
+			array(
+				array(
+					'methods'  => WP_REST_Server::READABLE,
+					'callback' => array( $this, 'get_mailchimp_status' ),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Check if MailChimp is set up properly.
+	 *
+	 * @return bool
+	 */
+	private function is_connected() {
+		$option = get_option( 'jetpack_mailchimp' );
+		if ( ! $option ) {
+			return false;
+		}
+		$data = json_decode( $option, true );
+		if( ! $data ) {
+			return false;
+		}
+		return isset( $data['follower_list_id'], $data['keyring_id'] );
+	}
+	/**
+	 * Get the status of current blog's Mailchimp connection
+	 *
+	 * @return mixed
+	 * code:string (connected|unconnected),
+	 * connect_url:string
+	 * site_id:int
+	 */
+	public function get_mailchimp_status() {
+		$is_wpcom    = ( defined( 'IS_WPCOM' ) && IS_WPCOM );
+		$site_id     = $is_wpcom ? get_current_blog_id() : Jetpack_Options::get_option( 'id' );
```

Comment:
I would recommend returning an error if we can't get a site ID via `Jetpack_Options::get_option( 'id' )` (it could be because the site is not connnected properly, for example).

Reply:
cc @dmsnell who was Yesterday looking for a standard on how to get the `blog_id`

I proposed a filter, something like `apply_filter( 'get_current_wpcom_blog_id', Jetpack_Options::get_option( 'id' ) )` or maybe a `Jetpack::get_wpcom_blog_id` method

Reply:
Done in https://github.com/Automattic/jetpack/pull/11092/commits/77713de4312deee344ed4e67e92d7e9119c649df